### ISSUE TITLE
Use in combination with MudSystem

### ISSUE THREAD

**Issue by PixelFarm1** [Frvetz/FS22_SlipDestroysFields#15]
Hey!

First of all, very nice mod! 

I have been trying to use the mod in combination with the MudSystem mod from Viper. I wanted to get the combined effect that when the wheels sink, the ground is also destroyed due to slipping tires. However, it doesn't seem to work properly. As soon as the wheels sink below a certain level, the destruction comes to a halt, no matter how much I slip the wheels.

I would appreciate if you could have a look at the issue and give me a hint to what I could change myself or push a update to the repo or modhub.

Again, well done, I hope to hear from you! 🥇

**Comment by Frvetz** [comment]
Hey!
Thank you!

The first idea that would come to my mind would be that the vehicle must be on a field, due to the game engine, as the ground can only be influenced there. 

If you are on a field, I'll take a closer look at it tonight!

Cheers
Gerrit / frvetz
😃

**Comment by PixelFarm1** [comment]
Thanks for the quick reply. I am indeed on a field! The slip works fine until the machine has sunk into the ground. I thought it might not recognize the ground contact in that case, but from what I can understand (I'm a lua noob) you already handle the case of no ground contact? 😃

**Comment by Frvetz** [comment]
Oh okay. Yeah that might be the case. The script checks if the wheel as ground contact in order to prevent errors, so if the game thinks it has no contact when sinking in, nothing will be destroyed... I'll take a look at it tonight!

**Comment by PixelFarm1** [comment]
Nice! 😄 Would love to see if your mod could actively check for the mudsystem mod and if it is in a field and the mudsystem returns a sink value the slip is adjusted to always make slipping tracks if the machine has sunk to a certain level 👍

**Comment by Frvetz** [comment]
That's an awesome idea!!! Yeah I'll do that if the mud system has a value for that 👍

**Comment by PixelFarm1** [comment]
Check out the fieldSinkUpdater.lua from the mudsystem when you get to it. I guess the mfsu.fieldSinkAmount could be usable 😃

**Comment by Frvetz** [comment]
Oh thanks! But help me real quick 🤔 isn't the mud above the ground and therefore the destroyed field is invisible or do you want to remove the mud when the wheel slips?

**Comment by PixelFarm1** [comment]
There are two ways to look at it. Firstly, there is a mud filltype, I don't think you have to care about that since.. yeah, as you said if there is mud you can't see the tracks. However, it also has an effect from rain. If it rains, all fields turn into mud, for 10 minutes the sink level increases until it reaches max sink level. That is where I would love to see your mod come in and add further realism. Since, in fields there is none of the mud filltype unless the map creator has added it. 

TLDR: The mud system turns fields to mud when it rains. Then there is no filltype on top of the ground. You just sink into the field and that is when i'd like to see the wheels also made tracks when you start slipping

**Comment by Frvetz** [comment]
Ahhhhhh alright now I got it. Thank you!

**Comment by PixelFarm1** [comment]
Seems like this mod "listens" to the field wetness value from the mudSystem mod. Just leaving it here if you need some inspo 😄 https://www.farming-simulator.com/mod.php?lang=en&country=us&mod_id=228604&title=fs2022

**Comment by PixelFarm1** [comment]
I tested some more. If I am driving forwards constantly when the rain starts and don't stop, the slip destruction seem to work even when the machine sinks (I have set the time needed with rain to 2 minutes for mudSystem). But, if I come to a stop and start driving, it makes a slight slip, then stops although it looks like it should register slip.

**Comment by Frvetz** [comment]
Yeah sometimes the slip is not recognized... Most of the time when you stop and then start driving with slip. But there is nothing really that I can do to fix that so checking the mudSystem sink factor is best I think

**Comment by Frvetz** [comment]
Got it working without the need of variables from the mud system. So now other mods like REA will also affect the ground.
Tomorrow I'll add a game setting for turning it off and then update the file here 😃 

Thank you for the idea!!

**Comment by Frvetz** [comment]
Just uploaded the update 😃 https://github.com/Frvetz/FS22_SlipDestroysFields/releases/latest

**Comment by PixelFarm1** [comment]
Very nice! 😄 

Only downside to this is that it would be nice to be able to set a specific threshold value for the sink level where the destroy field part kicks in. I am Ok with this solution too but it might be an idea for the future? Example: when sink reaches 40% of max the destruction kicks in. Since the sink starts the moment it starts raining now we'll have a more unrealistic behavior 😃

**Comment by Frvetz** [comment]
Currently I have set it to 0.1 which is when the wheel is visibly in the ground but I can change it so that the player can decide the threshold 👍

**Comment by Frvetz** [comment]
✅ Now you can set the depth at which point the ground gets destroyed in the game settings 🙂 You can choose from 1.0 to 3.0 (0.5 steps). A value higher than 3.0 seems extremely unrealistic so I didn't go higher than that.

https://github.com/Frvetz/FS22_SlipDestroysFields/releases/latest
